util.c

```diff
@@ -363,3 +364,242 @@ int UTIL_ensure_path(const char *toDir)
+             "   POWER_PROC=./none\n"
+             "   PARAM=\n\n",
+             name, name, start_time, period, name, path);
+
+   for (curr = points; curr->id; curr++) {
+      int ignore = 0;
+      for (itr = head; itr; itr = itr->next) {
+         if (0 == strcasecmp(itr->name, curr->id)) {
+            ignore = 1;
+            break;
+         }
+      }
+
+      if (mode == 1) {
+         if (!ignore)
+            printf("%s\n", curr->id);
+      }
+      else {
+         if (!ignore) {
+            printf("   <SENSOR>\n");
+            printf("      NAME=%s\n", name);
+            printf("      SENSOR_KEY=%s\n", curr->id);
+            printf("      LOCATION=%s\n", curr->location);
+            printf("      GROUPS=%s\n", curr->group);
+            printf("   </SENSOR>\n");
+         }
+         else {
+            printf("   <IGNORE>\n");
+            printf("      SENSOR_KEY=%s\n", curr->id);
+            printf("   </IGNORE>\n");
+         }
+      }
+   }
+
+   if (header)
+      printf("</SUBPROCESS>\n");
+
+   return 0;
+
+usage:
+   printf("%s [-n <subprocess name>] [-H] [-p <period>] [-s <start time>] [-t <telemetry path] [-x <exclude tlm id>] [-x <>] ...\n",
+      argv[0]);
+   return 0;
+
+}
+
+int UTIL_print_sensor_metadata(struct UTILTelemetryInfo *points,
+   struct UTILEventInfo *events)
+{
+   struct UTILTelemetryInfo *curr;
+   struct UTILBitfieldInfo *bitr;
+   struct UTILEventInfo *eitr;
+   const char *type;
+
+   for (curr = points; curr->id; curr++) {
+      type = "SENSOR";
+      if (curr->computed_by)
+         type = "VSENSOR";
+
+      printf("<%s>\n", type);
+      printf("   KEY=%s\n", curr->id);
+      printf("   LOCATION=%s\n", curr->location);
+      printf("   SUBSYSTEM=%s\n", curr->group);
+      printf("   UNITS=%s\n", curr->units);
+      printf("   DIVISOR=%lf\n", curr->divisor);
+      printf("   OFFSET=%lf\n", curr->offset);
+      printf("   NAME=%s\n", curr->name);
+      printf("   DESCRIPTION=%s\n", curr->desc);
+      if (curr->computed_by)
+      printf("   FUNCTION=%s\n", curr->computed_by);
+
+      if (curr->bitfields) {
+         for (bitr = curr->bitfields; bitr->set_label; bitr++) {
+            printf("   <ENUM>\n");
+            printf("      VALUE=%u\n", bitr->value);
+            printf("      LABEL=%s\n", bitr->set_label);
+            if (bitr->clear_label)
+               printf("      UNSET=%s\n", bitr->clear_label);
+            printf("   </ENUM>\n");
+         }
+      }
+
+      printf("</%s>\n", type);
+   }
+
+   for (eitr = events; eitr->port_name; ++eitr) {
+      printf("<EVENT>\n");
+      printf("   PORT_NAME=%s\n", eitr->port_name);
+      printf("   PORT=%u\n", socket_get_addr_by_name(eitr->port_name));
+      printf("   ID=%u\n", eitr->id);
+      printf("   NAME=%s\n", eitr->name);
+      printf("   DESCRIPTION=%s\n", eitr->desc);
+      printf("</EVENT>\n");
+   }
+
+   return 0;
+}
+
+int UTIL_print_html_telem_dict(struct UTILTelemetryInfo *points,
```

Comment:
Instead of trying to generate a html website from libproc, would it maybe make more sense to output a simpler format, like our existing configuration file format (since you already added it) or JSON and use a tool like pandoc to convert that basic format into a website?

Reply:
I agree.  Adding in specific presentation markup generation to the library is not a good decision.  Unfortunately the datalogger configuration file isn't exactly spec XML, so I'm leaning toward JSON.  I'll update the pull request shortly.

Reply:
I just pushed an update to the branch